### PR TITLE
[SYCL][COMPAT] Add math extend_vcompare[2/4] to SYCLCompat

### PR DESCRIPTION
**Pull request by OuadiElfarouki**
This PR adds math `extend_vcompare[2/4] `operators (4 in total) along with unit-tests for signed and unsigned int32 cases.
Also, Unit-tests from previous `extend_v*4`  #14078 and `extend_v*2` #13953 are moved to two different files.

### CODE REVIEW

File: sycl/doc/syclcompat/README.md [hunk 1]
```diff
@@ -1855,6 +1855,244 @@ inline constexpr RetT extend_max_sat(AT a, BT b, CT c,
...
+/// \tparam [in] AT The type of the first value, can only be 32 bit integer
+/// \tparam [in] BT The type of the second value, can only be 32 bit integer
+/// \param [in] a The first value
+/// \param [in] b The second value
+/// \param [in] c The third value
+/// \returns The extend vectorized average of the two values with saturation
+template <typename RetT, typename AT, typename BT>
+inline constexpr RetT extend_vavrg2_sat(AT a, BT b, RetT c);
+```
+
```
Comment (Alcpz) [inline sycl/doc/syclcompat/README.md:2095]: New functions missing in README

File: sycl/include/syclcompat/math.hpp [hunk 2]
```diff
@@ -1222,4 +1334,600 @@ inline constexpr RetT extend_shr_sat_wrap(T a, uint32_t b, uint32_t c,
...
+/// \param [in] c The third value
+/// \returns The extend vectorized average of the two values with saturation
+template <typename RetT, typename AT, typename BT>
+inline constexpr RetT extend_vavrg2_sat(AT a, BT b, RetT c) {
+  return detail::extend_vbinary2<RetT, true, false>(a, b, c, detail::average());
+}
+
+/// Extend \p a and \p b to 33 bit and vectorized compare input values using
+///
+/// specified comparison \p cmp .
```
Comment (Alcpz) [inline sycl/include/syclcompat/math.hpp:1604]:
```suggestion
/// specified comparison \p cmp .
///
```

File: sycl/include/syclcompat/math.hpp [hunk 3]
```diff
@@ -1222,4 +1334,600 @@ inline constexpr RetT extend_shr_sat_wrap(T a, uint32_t b, uint32_t c,
...
+/// \param [in] c The third value
+/// \returns The extend vectorized average of the two values with saturation
+template <typename RetT, typename AT, typename BT>
+inline constexpr RetT extend_vavrg4_sat(AT a, BT b, RetT c) {
+  return detail::extend_vbinary4<RetT, true, false>(a, b, c, detail::average());
+}
+
+/// Extend \p a and \p b to 33 bit and vectorized compare input values using
+///
+/// specified comparison \p cmp .
```
Comment (Alcpz) [inline sycl/include/syclcompat/math.hpp:1902]:
```suggestion
/// specified comparison \p cmp .
///
```

File: sycl/test-e2e/syclcompat/math/math_extend_v.cpp [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
+ *  SYCLcompat API
+ *
+ *  math_extend_v.cpp
+ *
+ *  Description:
+ *    math extend-vectorized helpers tests
+ **************************************************************************/
+
+// ===----------- math_extend_vfunc[2/4].cpp ---------- -*- C++ -*
+// --------------===//
```
Comment (Alcpz) [inline sycl/test-e2e/syclcompat/math/math_extend_v.cpp:24]: NIT: adjust to avoid splitting the comment

**Review by Alcpz — CHANGES_REQUESTED** [review]
Very minor stuff in the docs, otherwise LGTM, The review assumes that `extend_vbinary2` and `extend_vbinary4` are gonna be reviewed in their respective PRs.

**Review by Alcpz — APPROVED** [review]

**Review by yihanwg — APPROVED** [review]
LGTM

**Comment by OuadiElfarouki** [comment]
Due to the timeout of the `math_extend_v` tests causing the CI to fail, I just broke it down to 2 separate test files (one for each vectorization size).

**Comment by OuadiElfarouki** [comment]
@intel/llvm-gatekeepers PR should be ready! Thanks.

**Pull request merged**